File: 792-binary-search/binary-search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        #low and high defined for indexes
        low = 0
        high = len(nums) - 1

        while low <= high:
            mid = (low + high) // 2

            if nums[mid] == target:
                return mid
            if nums[mid] < target:
                low = mid + 1
            elif nums[mid] > target:
                high = mid -1
        return -1

File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int low_end = 0;
        int high_end = nums.length - 1;

        System.out.println(high_end);

        while (low_end <= high_end ){
            int mid = (low_end + high_end)/ 2;
            if (nums[mid] == target){
                return mid;
            } 

            if (nums[mid] < target){
                low_end = mid + 1 ;
            } else if (nums[mid] > target) {
                high_end = mid - 1 ;
            } 
        }
        return -1;
    }
}